shm/src/Game.cpp

```diff
@@ -86,11 +76,14 @@ void Game::printMap() {
     }
 }
 
-void Game::selectOption() {
-    printWelcomeScreen();
+Game::MenuOption Game::selectOption() {
+    printIntenface();
     printMenu();
     size_t option {};
-    std::cin >> option;
+    do {
+        std::cout << "Please insert you choice: ";
+        std::cin >> option;
+    } while (validatingMenuChoose(option) == false);
     menuOption_ = static_cast<MenuOption>(option);
     switch(menuOption_) {
```

Comment:
'switch' should be outside 'selectOption' method which should only return one option chosen by the player to be used by 'switch' later. The only option checked from 'selectOption' method is MenuOption::Exit (in 'startGame' loop). 'switch' should be inside a separate method called for example 'makeAction' which accepts there enum returned form 'selectOption'.